File: vt06.java
import java.util.Scanner;
/*
Hailun
*/
public class Main {

    public static void main(String[] args) {

        try (Scanner s = new Scanner(System.in)) {

            int n = s.nextInt();

            int count = 0;
            double sum = 0;

            for (int i = 0; i < n; i++) {

                long e = s.nextLong();
                if(Math.abs(e) % 2 == 1) {
                    sum += e;
                    count++;
                }

            }


            System.out.printf("%.4f" , (sum / count));

        }
    }

}

File: vt06.cpp
/*Hailun*/
#include "iostream"
#include "iomanip"
#define MAX 512345
using namespace std;
void NhapM(int a[], int n)
{
    for(int i=0; i<n; i++)
        cin >> a[i];
}

int TongLe(int a[], int n)
{
    int sum = 0;
    for(int i=0; i<n; i++)
    {
        if(a[i] % 2 != 0)
            sum+=a[i];
    }
    return sum;
}

int DemLe(int a[], int n)
{
    int count = 0;
    for(int i=0; i<n; i++)
    {
        if(a[i] % 2 != 0)
            count++;
    }
    return count;
}

int main()
{
    int a[MAX], n;
    cin >> n;
    NhapM(a,n);
    cout << fixed << setprecision(4) << (float)TongLe(a,n)/DemLe(a,n) << endl;
    return 0;
}
